libctru/source/services/cam.c

```diff
@@ -0,0 +1,737 @@
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_SetPackageParameterWithContext(CAMU_PackageParameterContext param) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x00340140;
+    *(CAMU_PackageParameterContext*) cmdbuf[1] = param;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_SetPackageParameterWithContextDetail(CAMU_PackageParameterContextDetail param) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x003501C0;
+    *(CAMU_PackageParameterContextDetail*) cmdbuf[1] = param;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_GetSuitableY2rStandardCoefficient(Y2R_StandardCoefficient* coefficient) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x00360000;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    *coefficient = (Y2R_StandardCoefficient) cmdbuf[2];
+    return cmdbuf[1];
+}
+
+Result CAMU_PlayShutterSound(CAMU_ShutterSoundType sound) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x00380040;
+    cmdbuf[1] = sound;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_DriverInitialize() {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x00390000;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_DriverFinalize() {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x003A0000;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_GetActivatedCamera(CAMU_CameraSelect* select) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x003B0000;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    *select = (CAMU_CameraSelect) cmdbuf[2];
+    return cmdbuf[1];
+}
+
+Result CAMU_GetSleepCamera(CAMU_CameraSelect* select) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x003C0000;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    *select = (CAMU_CameraSelect) cmdbuf[2];
+    return cmdbuf[1];
+}
+
+Result CAMU_SetSleepCamera(CAMU_CameraSelect select) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x003D0040;
+    cmdbuf[1] = select;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
+
+Result CAMU_SetBrightnessSynchronization(bool brightnessSynchronization) {
+    Result ret = 0;
+    u32* cmdbuf = getThreadCommandBuffer();
+    cmdbuf[0] = 0x003E0040;
+    cmdbuf[2] = brightnessSynchronization;
+
+    if ((ret = svcSendSyncRequest(camHandle)) != 0) return ret;
+    return cmdbuf[1];
+}
```

Comment:
Gotcha, I'll be making some touch-ups when I get the chance tomorrow. (e.g. Another thing I realized was that I could've just specified bit size for enum fields in structs instead of typing them as u8)